### PR TITLE
Performance test prototype

### PR DESCRIPTION
**Pull request by zyrolasting**
This revision adds Unix50 and EuroSys one liner performance tests, using `-m`.

As of this writing, `memstar.ndr.md` is now configured to use the `workers/run-perf.sh` script, and `ci.pash.ndr.md` are now listening on port 2048 to manage `memstar.ndr.md`. Go ahead and run `curl ci.pash.ndr.md:2048/stdout` to see where the tests are right now. :)

If you wish to run this yourself, check the README in `scripts/remote`.

Notes:

* @angelhof: The Dockerfile has been updated to declare a locale, fixing #106. I ended up needing to do this to get past more and more Python errors when running scripts. Since we discussed it before, can you help me verify that this will not negatively impact artifacts?
* I updated all worker scripts to use empty directories for locks, since `mkdir` is atomic.
* The correctness test configuration has been updated to use less code, since the performance test controller needed to share some functionality.
* @nvasilakis: Despite what we discussed, I ended up downloading the entire results directory. It turned out to be pretty fast (< 1s) for me, and since it's using `rsync` I trust that the controller will download only incremental changes from then on. Let me know if this is okay with you.

### CODE REVIEW

File: scripts/Dockerfile [hunk 1]
```diff
@@ -13,3 +13,9 @@ RUN sudo bash scripts/install.sh -p
 ENV PASH_TOP=/pash
 ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/usr/local/lib/"
 CMD ["/bin/bash"]
+
+RUN apt-get update
+RUN apt-get install -y locales locales-all
```
Comment (angelhof) [inline scripts/Dockerfile:18]: Nit: Maybe these should be moved with the `apt-get`s above

File: scripts/remote/workers/run-perf.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+if [ -d lock ]; then
+    echo "Busy on existing job."
+else
+    mkdir lock
+    trap 'rm -r lock' EXIT
+
+    docker exec $CONTAINER_NAME /bin/bash -c "cd /pash && git pull"
+    run ./execute_eurosys_one_liners.sh -m
+    run ./execute_unix_benchmarks.sh -m
```
Comment (angelhof) [inline scripts/remote/workers/run-perf.sh:30]: Does `execute_unix_benchmarks.sh` actually support `-m`? I think it just supports `-s` and `-l` :)

**Review by angelhof — CHANGES_REQUESTED** [review]
Thanks a lot! I reviewed the parts of the code that I understand (so I didn't check the ci infrastructure changes in detail). 

Regarding the unix benchmarks, I think that there is no `-m` flag. Did you change the script to support one?

Except for that (and the Dockerfile nit), everything looks good!

**Comment by angelhof** [comment]
> @angelhof: The Dockerfile has been updated to declare a locale, fixing #106. I ended up needing to do this to get past more and more Python errors when running scripts. Since we discussed it before, can you help me verify that this will not negatively impact artifacts?

Since you were able to run the perf tests it should be fine!

**Comment by zyrolasting** [comment]
@angelhof Requested changes are in. Thanks!

**Review by nvasilakis — APPROVED** [review]
Awesome! I haven't tested, I'm assuming we're firing this on beta/memstar for a few days and monitor the results:-)

**Comment by nvasilakis** [comment]
> * @nvasilakis: Despite what we discussed, I ended up downloading the entire results directory. It turned out to be pretty fast (< 1s) for me, and since it's using `rsync` I trust that the controller will download only incremental changes from then on. Let me know if this is okay with you.

Sounds good!

**Review by angelhof — APPROVED** [review]

**Pull request merged**